components/nodemanager-service/pgdb/tags.go

```diff
@@ -5,41 +5,65 @@ import (
 
 	"github.com/chef/automate/components/compliance-service/api/common"
 	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
 )
 
+const sqlInsertNodeTag = `
+INSERT INTO nodes_tags
+(node_id, tag_id)
+VALUES ($1, $2)
+ON CONFLICT (node_id, tag_id)
+DO NOTHING;
+`
+
+const sqlInsertTag = `
+INSERT INTO tags
+(id, key, value)
+VALUES ($1, $2, $3)
+ON CONFLICT (key, value)
+DO UPDATE SET value=$3 RETURNING id;
+`
+
+// Note: ^^ this is silly. apparently you can't do an ON CONFLICT
+// DO NOTHING and still return the id, and we need to know the id b/c we still
+// need it to do the nodes_tags association. So we do a silly update so we can get
+// the id back from postgres :/ (vj)
+
 func (trans *DBTrans) addTags(tags []*common.Kv) ([]string, error) {
 	tagIDs := make([]string, 0, len(tags))
-	tagArr := make([]interface{}, 0, len(tags))
 
 	for _, keyValue := range tags {
-		tag := tag{
-			ID:    createUUID(),
-			Key:   keyValue.Key,
-			Value: keyValue.Value,
+		id := createUUID()
+		rows, err := trans.Query(sqlInsertTag, id, keyValue.Key, keyValue.Value)
```

Comment:
I *think* you can replace `Query` with `SelectStr` and it'll return the id string straight away without having to mess with the row.

Reply:
ooooh that would be nicer! ill give it a try/look it up